pkg/conn/basedb.go

```diff
@@ -52,15 +52,13 @@ func (d *defaultDBProvider) Apply(config config.DBConfig) (*BaseDB, error) {
 		if rawCfg.WriteTimeout != "" {
 			dsn += fmt.Sprintf("&writeTimeout=%s", rawCfg.WriteTimeout)
 		}
-		maxIdleConn = rawCfg.MaxIdleConns
+		maxIdleConns = rawCfg.MaxIdleConns
 	}
 	db, err := sql.Open("mysql", dsn)
 	if err != nil {
 		return nil, terror.ErrDBDriverError.Delegate(err)
 	}
-	if maxIdleConn >= 0 {
-		db.SetMaxIdleConns(maxIdleConn)
-	}
+	db.SetMaxIdleConns(maxIdleConns)
```

Comment:
you need to update the comment in _config/subtask.go_ L70~L72 now.

Reply:
Done in https://github.com/pingcap/dm/pull/266/files#diff-94b1e02e1fe1bd32d9eee7a40a730b00R69